examples/gno.land/p/demo/forms/forms.gno

```diff
@@ -0,0 +1,131 @@
+package forms
+
+import (
+	"errors"
+	"std"
+	"time"
+
+	"gno.land/p/demo/seqid"
+)
+
+type Field struct {
+	Label string
+
+	/*
+		string: "string";
+		number: "number";
+		boolean: "boolean";
+		choice: "['Pizza', 'Schnitzel', 'Burger']";
+		multi-choice: "{'Pizza', 'Schnitzel', 'Burger'}";
+	*/
+	FieldType string
+
+	Required bool
+}
+
+type Form struct {
+	ID    string
+	Owner std.Address
+
+	Title       string
+	Description string
+	Fields      []Field
+
+	CreatedAt time.Time
+	openAt    *time.Time
+	closeAt   *time.Time
+}
+
+type Submission struct {
+	FormID string
+
+	Author std.Address
+
+	/* ["Alex", 21, true, 0, [0, 1]] */
+	Answers string // json
+
+	SubmittedAt time.Time
+}
+
+type FormDatabase struct {
+	Forms []*Form
+
+	Answers []*Submission
+
+	IDCounter seqid.ID
+}
+
+func NewDatabase() *FormDatabase {
+	return &FormDatabase{
+		Forms:   make([]*Form, 0),
+		Answers: make([]*Submission, 0),
+	}
+}
+
+func (form *Form) IsOpen() bool {
+	openAt, errOpen := form.OpenAt()
+	closedAt, errClose := form.CloseAt()
+
+	noOpenDate := errOpen != nil
```

Comment:
Why not make an `errors.gno` file which will define these errors properly and then the code can check against them?

Reply:
not sure I understand this one, i pushed https://github.com/gnolang/gno/pull/2604/commits/6e35760095386660694c5b040346b97e394f1da9 to mimic what you did in [your disperse PR](https://github.com/gnolang/gno/pull/1414/files#diff-df729dfee87aef95039ecc62eb10398fb82010982714644c5f4f2c2a9ea76bd9) but then i realised this isn't exactly what your comment was asking